registry/handlers/helpers.go

```diff
@@ -15,3 +19,38 @@ func closeResources(handler http.Handler, closers ...io.Closer) http.Handler {
 		handler.ServeHTTP(w, r)
 	})
 }
+
+// copyFullPayload copies the payload of a HTTP request to destWriter. If it
+// receives less content than expected, and the client disconnected during the
+// upload, it avoids sending a 400 error to keep the logs cleaner.
+func copyFullPayload(responseWriter http.ResponseWriter, r *http.Request, destWriter io.Writer, context ctxu.Context, action string, errSlice *errcode.Errors) error {
+	// Get a channel that tells us if the client disconnects
+	var clientClosed <-chan bool
+	if notifier, ok := responseWriter.(http.CloseNotifier); ok {
+		clientClosed = notifier.CloseNotify()
+	} else {
+		panic("the ResponseWriter does not implement CloseNotifier")
+	}
+
+	// Read in the data, if any.
+	copied, err := io.Copy(destWriter, r.Body)
```

Comment:
I would expect this pattern to look a little more like this:

```go
ctx, cancel := context.WithCancel(ctx)
defer cancel()
copied := make(chan error)
closed := notifier.CloseNotify() // resolved however

go func() {
  _, err := io.Copy(destWriter, r.Body)
  copied <- err
  // .. handle error
}

select {
case <-copied:
case <-closed:
case <-ctx.Done():
}
```

Maybe needs more thought...